infrastructure/stacks/capacity_management/scripts/database_populator/database_populator.py

```diff
@@ -90,110 +90,208 @@ def filter_empty_address_fields(address):
-            "identifier": [
-                {
-                    "use": "official",
-                    "value": str(row["Identifier"]),
-                }
-            ],
-            "active": "true",
-            "type": row["Type"],
-            "name": str(row["Name"]),
-            "telecom": telecom_numbers,
-            "Address": filtered_address,
-            "createdDateTime": formatted_datetime,
-            "partOf": str(row["partof"]),
-            "createdBy": "Admin",
-            "modifiedBy": "Admin",
-            "modifiedDateTime": formatted_datetime,
-        }
-
-        if not telecom_numbers:  # If there is no telecom data in the source file
-            schema.pop("telecom")  # Remove the telecom field from the schema
-        if not filtered_address:  # If there is no address data in the source file
-            schema.pop("Address")  # Remove the Address field from the schema
-        if row["partof"] in [
-            "",
-            None,
-            "N/A",
-            "N/A",
-            "NO ID",
-        ]:  # If the partOf field is blank, null, n/a, N/A, or NO ID
-            schema.pop("partOf")  # Remove the partOf field from the schema
-
+        return transpose_organisations(row, formatted_datetime)
     elif table_name == "locations":
-        schema = {
-            "resourceType": "Location",
-            "id": str(row["Identifier"]),
-            "status": "Active",
-            "name": row["Name"],
-            "address": row["Address"],
-            "latitude": str(Decimal(row["Latitude"])),
-            "longitude": str(Decimal(row["Longitude"])),
-            "createdBy": row["CreatedBy"],
-            "createdDateTime": row["CreatedDateTime"],
-            "modifiedBy": row["ModifiedBy"],
-            "modifiedDateTime": row["ModifiedDateTime"],
-        }
+        return transpose_locations(row, formatted_datetime)
+    else:
+        raise ValueError("Unsupported table name: {}".format(table_name))
+
+
+def transpose_organisation_affiliations(row, formatted_datetime):
+    schema = {
+        "resourceType": "OrganizationAffiliation",
+        "id": str(row["Identifier"]),
+        "active": "true",
+        "identifier": [
+            {
+                "value": str(row["Identifier"]),
+            }
+        ],
+        "participatingOrganization": str(
+            row["FHIRParticipatingOrganizationnIdentifier"]
+        ),
+        "organization": row["FHIROrganizationIdentifier"],
+        "healthcareService": row["FHIRHealthcareServiceIdentifier"],
+        "location": row["FHIRLocationIdentifier"],
+        "createdBy": "Admin",
+        "createdDateTime": formatted_datetime,
+        "modifiedBy": "Admin",
+        "modifiedDateTime": formatted_datetime,
+    }
+
+    # Create a list of the keys in the schema
+    schema_keys = list(schema.keys())
+
+    # Remove empty keys from the schema
+    for key in schema_keys:
+        if (
+            schema[key] is None
```

Comment:
Quick question to clarify for me - here we have NO ID as an option. Elsewhere we have N/A and 0.